### PR TITLE
Docs review

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
Complements #7126 by fixing what I can fix.

## Todo:

- [x] Remove `_extension_name` from ESM examples
- [x] Document you can create layouts with `AnyWidgetComponent` via `get_child`.
- [x] Update layout guides to include `AnyWidgetComponent`.
- [x] Add a `MarioButton` tutorial based on `AnyWidgetComponent` or `ReactComponent`.
- [x] Philipp? Fix React issue identified below.

## Not

- [ ] https://github.com/holoviz/panel/issues/7087
- [ ] Make `ReactComponent` or `AnyWidgetComponent` callback examples https://holoviz-dev.github.io/panel/how_to/custom_components/esm/callbacks.html
- [ ] Add `AnyWidget` example to https://holoviz-dev.github.io/panel/how_to/custom_components/esm/dataframe.html
- [ ] Add `JSComponent` and `AnyWidget` versions of 
  - [ ] https://holoviz-dev.github.io/panel/how_to/custom_components/examples/esm_canvas.html

### CODE REVIEW

**Comment by MarcSkovMadsen** [comment]
I've removed the `_extension_name` from an ESM example. I don't believe its a thing for ESM components. It was for `ReactiveHTML`.

image

File: examples/reference/custom_components/AnyWidgetComponent.ipynb [hunk 1]
```diff
@@ -306,9 +305,182 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "91a8cb50-c38d-4d8a-b992-7bf013926a73",
+   "metadata": {},
+   "source": [
     "### React\n",
     "\n",
     "You can use React with `AnyWidget` as shown below."
```
Comment (MarcSkovMadsen) [inline examples/reference/custom_components/AnyWidgetComponent.ipynb:486]: The `React` example works on the server but not in the notebook

image

```python
import panel as pn
import param

from panel.custom import AnyWidgetComponent

pn.extension()


class CounterButton(AnyWidgetComponent):

    value = param.Integer()

    _importmap = {
        "imports": {
            "@anywidget/react": "https://esm.sh/@anywidget/react",
            "react": "https://esm.sh/react",
        }
    }

    _esm = """
    import * as React from "react"; /* mandatory import */
    import { createRender, useModelState } from "@anywidget/react";

    const render = createRender(() => {
      const [value, setValue] = useModelState("value");
      return (
        <button onClick={() => setValue(value + 1)}>
          count is {value}
        </button>
      );
    });
    export default { render }
    """

CounterButton().servable()
```

**Comment by MarcSkovMadsen** [comment]
I intended to fix broken links https://github.com/holoviz/panel/issues/7087 in this PR. But its grown. So I believe its better to get it reviewed now. Then I hope to fix broken links in another PR.

**Comment by philippjfr** [comment]
Merging as is for now.

**Review by philippjfr — APPROVED** [review]

**Pull request merged**